pkg/storage/indexheader/index/postings_test.go

```diff
@@ -100,3 +104,57 @@ func TestPostingValueOffsets(t *testing.T) {
 		})
 	}
 }
+
+func createPostingOffset(n int) []*indexheaderpb.PostingOffset {
+	offsets := make([]*indexheaderpb.PostingOffset, n)
+	for i := 0; i < n; i++ {
+		offsets[i] = &indexheaderpb.PostingOffset{Value: fmt.Sprintf("%d", i), TableOff: int64(i)}
+	}
+	return offsets
+}
+
+func Test_NewPostingOffsetTableFromSparseHeader(t *testing.T) {
+
+	testCases := map[string]struct {
+		existingOffsetsLen              int
+		postingOffsetsInMemSamplingRate int
+		protoSamplingRate               int64
+		expectedLen                     int
+		expectErr                       bool
+	}{
+		"downsample_noop_proto_has_equal_sampling_rate":           {100, 32, 32, 100, false},
+		"downsample_short_offsets":                                {2, 32, 16, 1, false},
+		"downsample_noop_short_offsets":                           {1, 32, 16, 1, false},
+		"downsample_proto_has_divisible_sampling_rate":            {100, 32, 16, 50, false},
+		"cannot_downsample_proto_has_no_sampling_rate":            {100, 32, 0, 0, true},
+		"cannot_upsample_proto_has_less_frequent_sampling_rate":   {100, 32, 64, 0, true},
+		"cannot_downsample_proto_has_non_divisible_sampling_rate": {100, 32, 10, 0, true},
+		"downsample_sampling_rates_ratio_does_not_divide_offsets": {33, 32, 16, 17, false},
+		"downsample_sampling_rates_ratio_exceeds_offset_len":      {10, 1024, 8, 1, false},
+		"downsample_sampling_rates_ratio_equals_offset_len":       {100, 100, 1, 1, false},
```

Comment:
can you put the fields on separate lines and add the field names? it's easier to read the test cases that way without having to look back at the test case definition. 

I normally do it for structs that have more than 2 fields